utils/loggers/wandb/wandb_utils.py

```diff
@@ -155,48 +165,74 @@ def __init__(self, opt, run_id=None, job_type='Training'):
-                                        job_type=job_type,
-                                        id=run_id,
-                                        allow_val_change=True) if not wandb.run else wandb.run
+            self.wandb_run = (
+                wandb.init(
+                    config=opt,
+                    resume="allow",
+                    project="YOLOv5"
+                    if opt.project == "runs/train"
+                    else Path(opt.project).stem,
+                    entity=opt.entity,
+                    name=opt.name if opt.name != "exp" else None,
+                    job_type=job_type,
+                    id=run_id,
+                    allow_val_change=True,
+                )
+                if not wandb.run
+                else wandb.run
+            )
         if self.wandb_run:
-            if self.job_type == 'Training':
+            if self.job_type == "Training":
                 if opt.upload_dataset:
                     if not opt.resume:
-                        self.wandb_artifact_data_dict = self.check_and_upload_dataset(opt)
+                        self.wandb_artifact_data_dict = self.check_and_upload_dataset(
+                            opt
+                        )
 
                 if isinstance(opt.data, dict):
                     # This means another dataset manager has already processed the dataset info (e.g. ClearML)
                     # and they will have stored the already processed dict in opt.data
                     self.data_dict = opt.data
                 elif opt.resume:
                     # resume from artifact
-                    if isinstance(opt.resume, str) and opt.resume.startswith(WANDB_ARTIFACT_PREFIX):
-                        self.data_dict = dict(self.wandb_run.config.data_dict)
+                    if isinstance(opt.resume, str) and opt.resume.startswith(
+                        WANDB_ARTIFACT_PREFIX
+                    ):
+                        self.data_dict = {
+                            k.lstrip("data_dict_"): v
+                            for k, v in self.wandb_run.config.items()
+                            if k.startswith("data_dict_")
+                        }
                     else:  # local resume
                         self.data_dict = check_wandb_dataset(opt.data)
                 else:
                     self.data_dict = check_wandb_dataset(opt.data)
-                    self.wandb_artifact_data_dict = self.wandb_artifact_data_dict or self.data_dict
+                    self.wandb_artifact_data_dict = (
+                        self.wandb_artifact_data_dict or self.data_dict
+                    )
 
                     # write data_dict to config. useful for resuming from artifacts. Do this only when not resuming.
-                    self.wandb_run.config.update({'data_dict': self.wandb_artifact_data_dict}, allow_val_change=True)
+                    self.wandb_run.config.update(
```

Comment:
This overcomes the nested dict issue we face when we add artifacts paths as parts of a logged nested dict to the config.